tests/api_test.py

```diff
@@ -1919,6 +1919,19 @@ def f(x1, x2, g):
     ):
       jax.vmap(f, (0, 0, None))(jnp.ones(2), jnp.ones(3), jnp.add)
 
+  def test_vmap_inconsistent_sizes_constructs_proper_error_message_kwargs(self):
+    # regression test for https://github.com/jax-ml/jax/issues/24406
+    def f(x1, x2, a3):
+      return x1 + x2 + a3
+
+    with self.assertRaisesRegex(
+      ValueError,
+      "vmap got inconsistent sizes for array axes to be mapped:\n"
+      r"  \* most axes \(2 of them\) had size 2, e.g. axis 0 of argument x1 of type float32\[2\];\n"
```

Comment:
We're seeing some test failures when this is run with `JAX_ENABLE_X64=true` becuase the default type of the arrays is `float64`, and so the error strings don't match.

you could address this via a more flexible regex, or by explicitly using `float32` dtypes below.

Reply:
Done